File: Backup 09-04-2021/API/API/API/Models/Comment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace API.Models
{
  public class Comment
  {
    [Key]
    public int CommentId { get; set; }
    public string CommentContent { get; set; }
    public int Rating { get; set; }
    public DateTime DatePost { get; set; }
    public Comment ParentComment { get; set; }
    public int? ParentCommentId { get; set; }
    public ICollection<Comment> ChildComments { get; set; }
    public Account Account { get; set; }
    [Required]
    public int AccountId { get; set; }
    public Course Course { get; set; }
    public int? CourseId { get; set; }
    public Lesson Lesson { get; set; }

    public int? LessonId { get; set; }
  }
}

File: Backup25-05-2021/src/app/model/comment.ts
export class Comment {
    commentId: Int32Array;
    commentContent: string;
    rating: number;
    datePost: string;
    type: string;
    accountId: Int32Array;
    courseId: Int32Array;
    lessonId: Int32Array;
    // commentId: Int32Array;
    // commentContent: string;
    // rating: number;
    // datePost: string;
    // type: string;
    // accountId: string;
    // courseId: string;
    // lessonId: string;
}
export class SubComment {
    subCommentId: Int32Array;
    subCommentContent: string;
    subRating: number;
    subDatePost: string;
    isLiked: boolean;
    parentCommentId: Int32Array;
}
